Sources/Basics/Observability.swift

```diff
@@ -268,6 +250,28 @@ public struct Diagnostic: CustomStringConvertible, Equatable {
         Self(severity: .error, message: message.description, metadata: metadata)
     }
 
+    public static func error(_ error: Error, metadata: ObservabilityMetadata? = .none) -> Self {
+        var metadata = metadata
+        // FIXME: this brings in the TSC API still
+        if let errorProvidingLocation = error as? DiagnosticLocationProviding, let diagnosticLocation = errorProvidingLocation.diagnosticLocation {
+            metadata = metadata ?? ObservabilityMetadata()
+            metadata?.legacyDiagnosticLocation = .init(diagnosticLocation)
+        }
+
+        let message: String
+        // FIXME: this brings in the TSC API still
+        // FIXME: string interpolation seems brittle
```

Comment:
I agree, but I think it suffices for incremental progress.  I think what we should do in the long run is to form the strings from the primary data at display time, where we know things like how wide the output can be and can choose long or short messages appropriately.  But that's pretty far in the future, but it would be great as a goal, I think.  So for now this is fine, then we can make it more robust, and finally maybe make it be completely parameterized and compute the actual message when rendering to console or UI.